src/main/java/org/reaktivity/rym/internal/commands/install/Install.java

```diff
@@ -0,0 +1,193 @@
+                // TODO Handle file not found
+                System.err.format("Error: %s is not an array of strings", DEPS_PROP_REPOSITORIES);
+                return false;
+            }
+            JsonArray repositoriesArr = (JsonArray)repositoriesEl;
+            for (int i = 0; i < repositoriesArr.size(); i++)
+            {
+                if (repositoriesArr.get(i).isJsonPrimitive())
+                {
+                    repositories.add(repositoriesArr.get(i).getAsString());
+                }
+                else
+                {
+                    repositories.add(repositoriesArr.get(i).toString());
+                }
+            }
+        }
+
+        JsonElement dependenciesEl = depsObj.get(DEPS_PROP_DEPENDENCIES);
+        if (dependenciesEl != null)
+        {
+            if (!dependenciesEl.isJsonObject())
+            {
+                // TODO Handle file not found
+                System.err.format("Error: %s is not a JSON object", DEPS_PROP_DEPENDENCIES);
+                return false;
+            }
+            JsonObject dependenciesObj = (JsonObject)dependenciesEl;
+            dependenciesObj.keySet().forEach(key ->
+            {
+                String artifact = key.indexOf(':') > 0 ? key : String.format("%s:%s", DEFAULT_GROUP_NAME, key);
+                JsonElement versionEl = dependenciesObj.get(key);
+                // TODO Process version here. e.g. Is it a range?
+                if (versionEl.isJsonPrimitive())
+                {
+                    dependencies.put(artifact, versionEl.getAsString());
+                }
+                else
+                {
+                    dependencies.put(artifact, versionEl.toString());
+                }
+            });
+        }
+        return true;
+    }
+
+    private void writeDepsLockFile()
+    {
+        System.out.format("Generating lock file %s\n", DEFAULT_DEPENDENCY_LOCK_FILENAME);
+        JsonObject depsLockObj = new JsonObject();
+
+        JsonArray repositoriesArr = new JsonArray();
+        depsLockObj.add(DEPS_PROP_REPOSITORIES, repositoriesArr);
+        repositories.forEach(r -> repositoriesArr.add(r));
+
+        JsonObject dependenciesObj = new JsonObject();
+        depsLockObj.add(DEPS_PROP_DEPENDENCIES, dependenciesObj);
+        dependencies.forEach((dep, ver) ->
+        {
+            JsonElement versionEl = new JsonPrimitive(ver);
+            dependenciesObj.add(dep, versionEl);
+        });
+
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        try (BufferedWriter br = new BufferedWriter(new FileWriter(DEFAULT_DEPENDENCY_LOCK_FILENAME)))
+        {
+            br.write(gson.toJson(depsLockObj));
+        }
+        catch (IOException e)
+        {
+            // TODO Handle error
+            e.printStackTrace();
+        }
+
+    }
+
+    @Override
+    public void run()
+    {
+        if (!readDepsFile())
+        {
+            return;
+        }
+
+        writeDepsLockFile();
```

Comment:
Let exceptions escape for both these methods, and handle them here in `run()` instead.

That will let the methods themselves concentrate on the valid path forward instead of trying handle each and every exception so close to the point of origin which makes the code much harder to read.